core/includes/backup/backup.mysql.inc

```diff
@@ -0,0 +1,583 @@
+      $all_views = $this->getViews();
+
+      foreach ($all_tables as $table) {
+        if ($this->timeoutCheck()) {
+          return FALSE;
+        }
+        if ($table['name'] && !isset($exclude[$table['name']])) {
+          $file->write($this->getTableStructureSql($table));
+          $lines++;
+          if (!in_array($table['name'], $nodata)) {
+            $lines += $this->dumpTableDataSqlToFile($file, $table);
+          }
+        }
+      }
+      foreach ($all_views as $view) {
+        if ($this->timeoutCheck()) {
+          return FALSE;
+        }
+        if ($view['name'] && !isset($exclude[$view['name']])) {
+          $file->write($this->getViewCreateSql($view));
+        }
+
+      }
+      $file->write($this->getSqlFileFooter());
+      $file->close();
+      return $lines;
+    }
+    else {
+      return FALSE;
+    }
+  }
+
+  /**
+   * Backup the databases to a file using the mysqldump command.
+   */
+  private function backupToFileMysqldump(BackupFile $file) {
+    $nodata_tables = array();
+    $all_tables = $this->getTables();
+    $settings = $this->settings;
+
+    $command = 'mysqldump --result-file=%file --opt -Q --host=%host --port=%port --user=%user --password=%pass %name';
+    $args = array(
+      '%file' => $file->filepath(),
+      '%host' => $this->settings['database_host'],
+      '%port' => $this->settings['database_port'],
+      '%user' => $this->settings['database_user'],
+      '%pass' => $this->settings['database_pass'],
+      '%name' => $this->settings['database_name'],
+    );
+
+    // Ignore the excluded and no-data tables.
+    $db = $this->settings['database_name'];
+    if (!empty($settings['exclude_tables'])) {
+      foreach ((array) $settings['exclude_tables'] as $table) {
+        if (isset($all_tables[$table])) {
+          $command .= ' --ignore-table=' . $db . '.' . $table;
+        }
+      }
+    }
+    if (!empty($settings['nodata_tables'])) {
+      foreach ((array) $settings['nodata_tables'] as $table) {
+        if (isset($all_tables[$table])) {
+          $nodata_tables[] = $table;
+          $command .= ' --ignore-table=' . $db . '.' . $table;
+        }
+      }
+    }
+    // @todo: This is not yet supported.
+    $success = backup_migrate_exec($command, $args);
+
+    // Get the nodata tables.
+    if ($success && !empty($nodata_tables)) {
+      $tables = implode(' ', array_unique($nodata_tables));
+      $command = "mysqldump --no-data --opt -Q --host=%host --port=%port --user=%user --password=%pass %name $tables >> %file";
+      $success = backup_migrate_exec($command, $args);
+    }
+    return $success;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function restoreDatabaseFromFile(BackupFile $file) {
+    $settings = $this->settings;
+    $lines_executed = 0;
+
+    if ($file->open() && $connection = $this->getDatabaseConnection()) {
+      // Optionally drop all existing tables.
+      if (!empty($settings['utils_drop_all_tables'])) {
```

Comment:
I'm not totally sure if it can/should be removed; or how we would surface it in the UI. It possible that a new database table could be created in an update hook and not dropping it could cause an issue.

Reply:
I went ahead and removed this functionality for the time being. It's probably too risky to enable by default and has a higher risk of data loss. We can look at putting it back if it becomes a problem, but in a catastrophic event, if an incorrect table was restored its relatively easy to drop an extra table than to get back a bunch of tables that were incorrectly dropped.